### ISSUE TITLE
"kid" invalid, unable to lookup correct key in /var/app/vendor/firebase/php-jwt/src/JWT.php:434

### ISSUE THREAD

**Issue by gweller22** [googleapis/google-api-php-client#2242]
#### Environment details

  - OS: Ubuntu
  - PHP version: 8.1.2
  - Package name and version: google/apiclient v2.12.2

#### Steps to reproduce

  1. Composer update to latest release ( firebase/php-jwt:^6.0 )
  2. Run example code https://github.com/googleapis/google-api-php-client/blob/main/examples/idtoken.php
  3. On verifyIdToken UnexpectedValueException of `"kid" invalid, unable to lookup correct key`

#### Code example

```php
//Set the google client
$client = new Google\Client();
$client->setClientId('XXXXXXXXX');
$client->setClientSecret('XXXXXXXXX');

$redirect_uri = 'https://' . $_SERVER['HTTP_HOST'] . str_replace('?' . $_SERVER['QUERY_STRING'], '', $_SERVER['REQUEST_URI']);

//Add scope of user information
$client->addScope(['email']);
$client->setRedirectUri($redirect_uri);

if (isset($_GET['code']))
{
    $token = $client->fetchAccessTokenWithAuthCode($_GET['code']);

    $client->setAccessToken($token);

    $token_data = $client->verifyIdToken();

    // store in the session also
    print_r($token_data);
}
else
{
    $authUrl = $client->createAuthUrl();

    echo "<a class='login' href='" . $authUrl . "'>Connect Me!</a>";
}
```

#### Issue Information

- https://github.com/firebase/php-jwt/releases/tag/v6.0.0 has breaking changes to the JWT::decode
- Need to implemente the key vertion to https://github.com/googleapis/google-api-php-client/blob/main/src/AccessToken/Verify.php#L102-L106

**Comment by megamk** [comment]
Having the same issue with PHP 7.2/Centos 7

**Comment by Baylie21** [comment]
Same issue on PHP 8.0.1/Linux. Was able to fix by downgrading to GoogleAPIClient version 2.11 in composer.json, which does require version 6 of firebase/php-jwt

**Comment by dany1980it** [comment]
Same here, docker/php8.0.1, test and prod, 
downgrading to 6.0 I get some other error.

**Comment by bshaffer** [comment]
This should now be fixed in [v2.12.3](https://github.com/googleapis/google-api-php-client/releases/tag/v2.12.3). Please update to the latest version and let me know if this fixes the problem please!

**Comment by megamk** [comment]
> This should now be fixed in [v2.12.3](https://github.com/googleapis/google-api-php-client/releases/tag/v2.12.3). Please update to the latest version and let me know if this fixes the problem please!

yes, it fixed the issue, thank you
